tests/test_norm_ops.py

```diff
@@ -141,22 +141,56 @@ def test_accuracy_layernorm(shape, dtype):
 )
 
 
-@pytest.mark.weight_norm_interface
+@pytest.mark.weight_norm
 @pytest.mark.parametrize("shape, dtype, dim", WEIGHT_NORM_SHAPE_DTYPE_DIM)
 def test_accuracy_weightnorm(shape, dtype, dim):
+    dim = dim % len(shape)
+    v = torch.randn(shape, dtype=dtype, device="cuda", requires_grad=True)
+    g = torch.randn(
+        [1 if i != dim else shape[i] for i in range(v.ndim)],
+        dtype=dtype,
+        device="cuda",
+        requires_grad=True,
+    )
+    reduce_size = v.numel() // shape[dim]
+
+    ref_v = to_reference(v, False)
+    ref_g = to_reference(g, False)
+    ref_w_out = torch._weight_norm(ref_v, ref_g, dim)
+    with flag_gems.use_gems():
+        res_w_out = torch._weight_norm(v, g, dim)
+    gems_assert_close(res_w_out, ref_w_out, dtype, reduce_dim=reduce_size)
+
+    res_w_grad = torch.randn(shape, dtype=dtype, device="cuda", requires_grad=True)
+    ref_w_grad = to_reference(res_w_grad, False)
+
+    ref_v_grad, ref_g_grad = torch.autograd.grad(
+        ref_w_out, (ref_v, ref_g), grad_outputs=ref_w_grad
+    )
+    res_v_grad, res_g_grad = torch.autograd.grad(
+        res_w_out, (v, g), grad_outputs=res_w_grad
+    )
+    gems_assert_close(res_v_grad, ref_v_grad, dtype, reduce_dim=reduce_size)
+    gems_assert_close(res_g_grad, ref_g_grad, dtype, reduce_dim=reduce_size)
+
+
+@pytest.mark.weight_norm_interface
+@pytest.mark.parametrize("shape, dtype, dim", WEIGHT_NORM_SHAPE_DTYPE_DIM)
+def test_accuracy_weightnorm_interface(shape, dtype, dim):
```

Comment:
`pytest test_norm_ops.py -m weight_norm_interface -s --ref=cpu` may cause `RuntimeError: "weight_norm_kernel" not implemented for 'Half'`.

Reply:
This error occurred in a native PyTorch operator, which does not support this type. It's not an error from the operator I provided

Reply:
> This error occurred in a native PyTorch operator, which does not support this type. It's not an error from the operator I provided

We fix this bug through `to_reference(True)`, so that fp16 can run through converting to fp64 when --ref=cpu.